src/sync.cpp

```diff
@@ -1109,11 +1105,9 @@ LocalNode* Sync::checkpath(LocalNode* l, string* localpath, string* localname, d
             {
                 LOG_verbose << "New file. FaType: " << fa->type << "  FaSize: " << fa->size << "  FaMtime: " << fa->mtime;
             }
-            delete fa;
             return NULL;
         }
 
-        delete fa;
         fa = client->fsaccess->newfileaccess(false);
```

Comment:
yes, the effect is exactly the same between those two syntax - I think the assignment looks a bit cleaner - the assign (which will resolve to a move-assign since copy is disallowed for unique_ptr) will first delete any ownership it currently has, and then take ownership of the FileAccess* passed in from the assignment.  Happy to change it though if you prefer.

Reply:
nah, it's ok. I didn't notice that was the behaviour for `operator=`. Apologies for the noise.